server/database.go

```diff
@@ -887,9 +887,9 @@ func (s *Server) setDatabaseLabels(ctx context.Context, labelsJSON string, datab
 // Try to get database driver using the instance's admin data source.
 // Upon successful return, caller MUST call driver.Close, otherwise, it will leak the database connection.
 func (s *Server) getAdminDatabaseDriver(ctx context.Context, instance *api.Instance, databaseName string) (db.Driver, error) {
```

Comment:
nit: move this to "driver builder" so that it doesn't belong to server package. This allows us to remove server dependency from other components such as activity manager.

Reply:
This method needs to locate the engine-specific bin dir. And those bin dir is installed by the server, thus seems like it has to depend on the server module.

Reply:
I got an idea, will attempt to refactor it later.

Reply:
Sure, let me finish my batch here, still some leftover needs cleanup